app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java

```diff
@@ -549,9 +547,12 @@ else if(resultCode == RESULT_OK){
                 if (resultCode == RESULT_CANCELED) {
```

Comment:
@ctsims What does the TODO comment above the getSession() call mean? Is there like some edge case when the session needs to be loaded in another way? I'm hoping that what was null here was actually just currentSession, and not currentSession.getNeededDatum(), because that seems like a scarier failure mode.